anoma/namada-interface
# Sdk: add claim rewards

Closes #679.

---

### Added
- Support claim rewards tx in SDK

## Testing

1. Stake some NAM.
2. Check that you have rewards available with `namadac rewards`.
3. In Namadillo, add `window.sdk = sdk; window.BigNumber = BigNumber;` and then in the console something like:
```
var built = await sdk.tx.buildClaimRewards(
  {
    token: sdk.nativeToken,
    feeAmount: BigNumber(0),
    gasLimit: BigNumber(20_000),
    chainId: "localnet.4918ab9bafd7c5a6ac91c",
    publicKey: "tpknam1qptrn64myunqr4847yq4cn0uwek5ecwc7eeexjfc5npmd5kmg6ex563n5as"
  },
  {
    validator: "tnam1q9vhfdur7gadtwx4r223agpal0fvlqhywylf2mzx",
    source: "tnam1qz4sdx5jlh909j44uz46pf29ty0ztftfzc98s8dx"
  }
);

var signed = await namada.sign({
  txType: built.tx.tx_type(),
  signer: "tnam1qz4sdx5jlh909j44uz46pf29ty0ztftfzc98s8dx",
  tx: {
    txBytes: built.tx.tx_bytes(),
    signingDataBytes: built.tx.signing_data_bytes(),
  },
  wrapperTxMsg: built.tx.wrapper_tx_msg()
});

var processed = await sdk.rpc.broadcastTx({
  wrapperTxMsg: built.tx.wrapper_tx_msg(),
  tx: signed
})
```
4. Refresh Namadillo and check that your balance has increased.
